### PR TITLE
565 - Fix some IdsPopup animation/placement bugs

### PR DESCRIPTION
**Pull request by EdwardCoyle**
**Explain the details for making this change. What existing problem does the pull request solve?**
This PR makes some general improvements to the hide/show/place methods in IdsPopup:
- Animation between calls to `place()` is now respected properly when `animated` attribute is true
- Dependency on IdsRenderLoop for show/hide has been removed in favor of use of `async/await` and waiting for CSS Transitions to complete when animations are enabled
- Removed extraneous awaits/promises/etc
- When `animated` is false, time-staggered waiting for CSS animations to complete is bypassed.  This makes for faster/instantaneous reaction from things like tooltips/menus on show/hide, compared to a slight delay that used to exist.

This PR doesn't address some of the calculation problems with placement that still need fixing as part of #565.

**Related github/jira issue (required)**:
Partially Addresses #565

**Steps necessary to review your pull request (required)**:
Pull/Build/Run... then test these scenarios:

- Open http://localhost:4300/ids-popup and make sure the Popup toggles and opens in the correct position (to the right of the button)
- Smoke test http://localhost:4300/ids-popup/test-sandbox.html and make sure all configurations make sense (behavior is identical to main but the "default" configuration is now to use coordinates with top/left alignment).  All toggles should work and auto-update the component state.
- Open http://localhost:4300/ids-lookup/ and open the first Lookup Modal using its trigger button.  When the modal opens, it should not be blurry (this component is not fully implemented so the selections may not behave as expected).
- Smoke test http://localhost:4300/ids-contextual-action-panel/ (its behavior should be identical to Main)
- Smoke test http://localhost:4300/ids-menu-button (also should be identical)
- Smoke test http://localhost:4300/ids-dropdown (especially test the Dropdowns in the middle, which had some clicking/usability bugs initially after these fixes.  All behavior should be identical to main)

One test was adjusted but otherwise all previous tests should pass

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.

### CODE REVIEW

**Review by andriivarhanov — COMMENTED** [review]
Looks good, except for the failed build the only issue I've found is `ids-date-picker` and `ids-time-picker` popup is floating on first open (recordings are attached). Happens because popup `alignTarget` `arrowTarget` `align` props were set after popup visibility. Fixed here https://github.com/infor-design/enterprise-wc/commit/3811ceac78777734af9079bbd999d905a18d18f7 . Let me know if I should cherry pick the commit to this PR

https://user-images.githubusercontent.com/5204821/159457725-ec924709-11d2-424a-b891-a4b345d0af93.mov


https://user-images.githubusercontent.com/5204821/159457744-c88c9cc2-8062-4c24-af1f-3534fed0fa2c.mov

**Review by tmcconechy — COMMENTED** [review]
Looks good - the only other thing i see thats a bit odd is on http://localhost:4300/ids-popup/test-sandbox.html

1) I think we should set "Click page to set coords" off by default. As it made my believe something was working oddly. (Should it have a button to set it?)
2) im seeing what im not sure is correct by setting the "Containment" area for example setting between body, idscontainer and test container and clicking center, center doesnt seem to center. All the options between left, right, top, bottom each dont seem to make sense? (with center target selected). I.E. not seeing how it gets centered? But maybe its right just a bit confusing?

File: src/components/ids-popup/ids-popup.js [hunk 1]
```diff
@@ -59,12 +56,6 @@ export default class IdsPopup extends Base {
     this.#setInitialState();
     this.shouldUpdate = true;
     this.#attachEventHandlers();
-
-    // Stagger visibility and initial placement, if applicable
-    window.requestAnimationFrame(() => {
```
Comment (danielortiz1982) [inline src/components/ids-popup/ids-popup.js:64]: @EdwardCoyle Good job removing the `RAF` cleaned up nice. 👍

File: src/assets/css/ids-popup/test-sandbox.css [hunk 2]
```diff
@@ -1,11 +1,27 @@
...
 .demo-popup {
   padding: 50px;
+  text-align: center;
+}
+
+/*
+For the purposes of the sandbox page, remove IdsThemeSwitcher from document flow.
+DO NOT DO THIS in normal application setups.
+*/
+ids-theme-switcher {
```
[the remark this answers is not in the document]
Comment (EdwardCoyle) [reply src/assets/css/ids-popup/test-sandbox.css:16]: I found if you take the element out some default styling on the form components like button/input goes missing... maybe due to theme colors being applied against mode/version attributes.  It's a bug in those components we may need to work out elsewhere.

The theme switcher normally takes up space like a block element should. Without these rules, the position of the Popup and all the targets gets messed up.

**Comment by EdwardCoyle** [comment]
@tmcconechy I'll uncheck the coordinate placement on next push.  

From what I can see the alignments all work as they should.  If you are in Coordinate mode (no align target set), and you try to change alignments at x:0/y:0, the containment will stop most alignments from appearing to occur because the Bleed check stops the Popup from moving away from the viewport's edge.  Uncheck "Bleeds" and you should see alignment occurring as if it's not hitting boundaries.  If you want to check that it happens with bleeds on, just set the coordinates to something closer to the middle of the page (try x:500 y:500 for example).  In coordinate mode, alignment is done by the corner of the popup as defined by those X/Y alignment values, so if "top, left" is applied, it'll use the top/left corner for its alignment, etc.

**Comment by EdwardCoyle** [comment]
@andriivarhanov That's a good find I hadn't checked before.  Feel free to push that one to this branch!

**Review by jmacaluso711 — COMMENTED** [review]
Aside from the failing Axe test all looks good to me

**Review by andriivarhanov — APPROVED** [review]

**Review by tmcconechy — APPROVED** [review]

**Pull request merged**